cmd/tracee-ebpf/internal/printer/printer.go

```diff
@@ -105,32 +120,40 @@ func (p tableEventPrinter) Preamble() {
 	fmt.Fprintln(p.out)
 }
 
-func (p tableEventPrinter) Print(event trace.Event) {
-	ut := time.Unix(0, int64(event.Timestamp))
+func (p tableEventPrinter) Print(event protocol.Event) {
+	evt, ok := event.Payload.(trace.Event)
+
+	if !ok {
+		if p.debug {
```

Comment:
I'm outputting to p.out here because this isn't really an error IMO. It's perfectly valid to output an event unrelated to the printer type, it's the printer's fault for not supporting it. As such I also guarded it under debug.

Reply:
It's like hiding logical errors that shouldn't normally happen and making them visible only if I enable the debug flag. Isn't it?

Reply:
Why is it a logical error though? Theoretically `protocol.Event`s can have any kind of payload, it's the printers job to either be completely agnostic to it or report if it's unable to print the type. As far as I see it it's a completely valid code path. 

This is partially why I was hesitant on changing the `Print` signature.

Reply:
Theoretically it's protocol.Event, but in practice tracee-ebpf is supposed to emit trace.Event. If that's not the case we should fail fast and catch such errors.

Reply:
Even so I don't think the printer is the place to fail on this (if it's something to fail on at all). Especially since the "table" usecase is just for data dumping. Other printers are perfectly fine not knowing the payload, for example the json and gob ones.
If we would have changed `cfg.ChanEvents` to `chan protocol.Event`, would you still be of the same opinion? Or if there were multiple output channels?

Reply:
Getting this error in printer is extremely unlikely to happen. And it's extremely hard to understand why we show error only if the debug flag is enabled. I'll also stop here and let another reviewer resolve this conversation.

Reply:
It's not an error it's just a limitation of the table printer. Imagine we would be outputting to table and in the meanwhile we would have background events which aren't `trace.Event`. Obviously we wouldn't want to print to table but there's nothing erroneous about the events happening, they are simply irrelevant to the logging method we chose. As such notifying the user they happened is, IMO, relevant only if he's debugging, otherwise the user probably wouldn't care at all about those events, and printing the error would just be a visual nuisance. 

Anyways, I think we both made our cases clear so I will also stop now and wait for another reviewer to resolve.